### PR TITLE
Move unbind operations from dtensor to shardtensor dispatch

### PR DESCRIPTION
**Pull request by coreyjadams**
# PhysicsNeMo Pull Request

This PR is a small slice of the ShardTensor refactor PR.  It takes one operation (unbind) that was build as a DTensor extension and rebuilds it as a shardtensor extension instead.  It also adds explicit testing of the unbind operations.

The unbind implementation on DTensor has API issues with PyTorch so I pulled this out so we can merge it faster, and so we don't get stuck in CI bugs while the ShardTensor PR is under review.  This has none of the actual API changes.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

**Comment by coreyjadams** [comment]
Closes #1570

File: test/domain_parallel/ops/test_unbind.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    original_tensor = torch.rand(shape, device=dm.device, requires_grad=backward)
+
+    sharded_tensor = scatter_tensor(
+        original_tensor,
+        global_src=0,
+        mesh=distributed_mesh,
+        placements=placements,
+        requires_grad=True,
+    )
```
[the remark this answers is not in the document]
Comment (coreyjadams) [reply test/domain_parallel/ops/test_unbind.py:71]: this isnt necessary in the tests

File: physicsnemo/domain_parallel/custom_ops/_tensor_ops.py [hunk 2]
```diff
@@ -16,130 +16,202 @@
...
-
 aten = torch.ops.aten
 
 
-@register_prop_rule(aten.unbind.int, schema_info=RuntimeSchemaInfo(1))
-def unbind_rules(op_schema: OpSchema) -> OutputSharding:
-    r"""Propagation rule for ``torch.unbind`` on ShardTensor.
+def _unbind_output_metadata(
+    input_spec: ShardTensorSpec, dim: int
+) -> tuple[int, list, dict[int, list[torch.Size]]]:
```
[the remark this answers is not in the document]
Comment (coreyjadams) [reply physicsnemo/domain_parallel/custom_ops/_tensor_ops.py:46]: No this isn't strictly correct.

**Review by pzharrington — APPROVED** [review]
